src/subscription.js

```diff
@@ -389,9 +389,14 @@ Subscription.prototype.close = function(callback) {
   clearTimeout(this.leaseTimeoutHandle_);
   this.leaseTimeoutHandle_ = null;
 
-  this.flushQueues_().then(function() {
-    self.closeConnection_(callback);
-  });
+  var flushPromise = this.flushQueues_();
+  if (callback) {
+    flushPromise.then(function() {
+      self.closeConnection_(callback);
+    });
+  } else {
+    return flushPromise;
+  }
```

Comment:
This should already be ensured by:

https://github.com/googleapis/nodejs-pubsub/blob/ace59b086c9f72fb1e582398f7cf232b1428b6bc/src/subscription.js#L1240-L1247